Markers dock

bmatherly opened the pull request:
Create a new dock panel to list all markers, allow sorting and editing. Clicking on a marker selects it and seeks the timeline to that marker.

![image](https://user-images.githubusercontent.com/821968/140596112-15ce7aa2-7260-4722-b459-b74c5277ae40.png)

Interested in comments on usability

ddennedy reviewed (changes requested):
I am still reviewing, but some initial feedback. There should be a shortcut for this. I suggest Ctrl+Shift+6 (because it continues the Ctrl+number theme and on US keyboard, carot is above 6). You can also add to the end of the main toolbar. I usually turn off text under icons and have plenty more room. Besides, it can overflow. I think we will leave this out of any of the preset layouts for now since that is a pain in the butt to maintain. I do like the default placement when I open it in the Editing layout. I did also test some functionality, and it is working good.

bmatherly commented on src/widgets/editmarkerwidget.cpp:
```diff
@@ -111,22 +134,29 @@ void EditMarkerWidget::on_colorButton_clicked()
+        int duration = m_endSpinner->value() - m_startSpinner->value() + 1;
+        m_durationLabel->setText(MLT.producer()->frames_to_time(duration));
+    } else {
+        m_durationLabel->setText("00:00:00:00");
```
I would like to, but this is the "else" case of "if (MLT.producer())". So I would need some other object to call frames_to_time() on. Maybe I could just clear out the duration field to be blank.

ddennedy replied:
Oh yeah, that makes sense. But I could not figure out how to make this appear. When I close a project with markers and one of the markers selected in the panel, it still shows the old info in the edit form within the panel. I think that is the last thing that needs to be addressed here.

ddennedy replied:
In the player, I use "--:--:--:--" to represent cleared or no timecode. Some languages use different numeric symbols than "0".

bmatherly commented:
> I suggest Ctrl+Shift+6 (because it continues the Ctrl+number theme and on US keyboard, caret is above 6).

Done

> You can also add to the end of the main toolbar. I usually turn off text under icons and have plenty more room. 

Done

bmatherly commented:
Sorry to pile on, but I added two additional features that I had been working on:

**Recent color selection in marker context menu:**

![image](https://user-images.githubusercontent.com/821968/141225713-4c6b0dba-645c-41a7-9d0d-ce29d93d6467.png)

**New menu in Marker panel to provide "Remove All" and Column hiding:**

![image](https://user-images.githubusercontent.com/821968/141226099-4f40255e-a4a7-461a-827b-0f0b9907f8fb.png)

ddennedy commented:
> Sorry to pile on

That's fine. I think the main thing people will miss here is a search box.

> Recent color selection in marker context menu

I think we can make that menu more narrow. Not too narrow tho because a translation for "Other" could be slighter longer than the English. But I think about half the current width is fine.

ddennedy commented:
I think this looks a little nicer
```
--- src/qml/modules/Shotcut/Controls/Marker.qml
+++ src/qml/modules/Shotcut/Controls/Marker.qml
@@ -82,10 +82,11 @@ Item {
         Menu {
             id: colorMenu
             title: qsTr('Color')
+            width: 100
             Instantiator {
                 model: markers.recentColors
                 MenuItem {
-                    implicitHeight: 14
+                    implicitHeight: otherMenuItem.implicitHeight
                     id: menuItem
                     background: Rectangle {
                         color: modelData
@@ -99,6 +100,7 @@ Item {
             }
             MenuSeparator { }
             MenuItem {
+                id: otherMenuItem
                 text: qsTr('Other...')
                 onTriggered: {
                     colorDialog.color = root.markerColor
```

bmatherly commented:
I took your suggestion on the color context menu size.

I also added a search box for the marker dock and converted the buttons to toolbar style to be consistent with other docks.

![image](https://user-images.githubusercontent.com/821968/141659600-01f8d06c-859d-48e1-9ad1-a9f0f538bb42.png)
